pkg/install/process.go

```diff
@@ -16,135 +16,62 @@ limitations under the License.
 package install
 
 import (
+	"context"
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
-	"os/exec"
-	"runtime"
-	"strings"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
 
-	"github.com/k0sproject/k0s/internal/util"
-	"github.com/mitchellh/go-ps"
-	"gopkg.in/yaml.v2"
+	config "github.com/k0sproject/k0s/pkg/apis/v1beta1"
+	"github.com/k0sproject/k0s/pkg/constant"
 )
 
 type K0sStatus struct {
-	Version  string
-	Pid      int
-	PPid     int
-	Role     string
-	SysInit  string
-	StubFile string
-	Output   string
+	Version       string
+	Pid           int
+	PPid          int
+	Role          string
+	SysInit       string
+	StubFile      string
+	Output        string
+	Workloads     bool
+	Args          []string
+	ClusterConfig *config.ClusterConfig
+	K0sVars       constant.CfgVars
 }
 
-func GetPid() (status *K0sStatus, err error) {
-	pid, ppid, err := getProcessID()
-	if err == nil && pid != nil {
-		status = &K0sStatus{
-			Pid:  *pid,
-			PPid: *ppid,
-		}
-		return status, nil
-	}
-	return &K0sStatus{}, nil
-}
-
-func getProcessID() (pid *int, ppid *int, err error) {
-	processList, err := ps.Processes()
-	if err != nil {
-		return nil, nil, err
-	}
-
-	for _, p := range processList {
-		if p.Executable() == "k0s" && hasChildren(p.Pid(), processList) {
-			pid := p.Pid()
-			ppid := p.PPid()
-			return &pid, &ppid, nil
-		}
-	}
-	return nil, nil, nil
-}
-
-func hasChildren(pid int, processes []ps.Process) bool {
-	for _, p := range processes {
-		if p.PPid() == pid {
-			return true
-		}
-	}
-	return false
-}
+func GetStatusInfo(runDir string, socket string) (status *K0sStatus, err error) {
```

Comment:
I'd prefer this to take a single socket path (or URI) instead of two args, runDir + socket.
```suggestion
func GetStatusInfo(socketPath string) (status *K0sStatus, err error) {
```